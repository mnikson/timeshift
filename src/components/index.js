import Header from './Header';
import Loading from './Loading';
import LocationInfo from './LocationInfo';
import PageLayout from './PageLayout';

export {
  Header,
  Loading,
  LocationInfo,
  PageLayout
};
