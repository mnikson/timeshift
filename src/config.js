require('dotenv').config();

const apiUrl = process.env.REACT_APP_API_SERVER_URL || 'http://localhost:4000';

module.exports = {
  /**
   * Root DOM element
   */
  rootElement: 'root',

  /**
   * Application name
   */
  applicationName: process.env.REACT_APP_APPLICATION_NAME || 'Timeshift',
  /**
   * API url
   */
  apiUrl
};
