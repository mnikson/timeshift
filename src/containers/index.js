/**
 * Containers main file
 */

import LocationsContainer from './LocationsContainer';

export {
  LocationsContainer
};
