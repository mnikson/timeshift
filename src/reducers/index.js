/**
 * Reducers index file
 */

// Libs
import { combineReducers } from 'redux';

// Reducers
import locations from './locations';

export default combineReducers({
  locations
});
