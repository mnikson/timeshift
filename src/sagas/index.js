/**
 * @file sagas
 * @description Sagas main fine
 */

// Libs
import { all, fork } from 'redux-saga/effects';

// Sagas
import location from './location';

export default function* rootSaga() {
  yield all([
    location
  ].map(fork));
}
