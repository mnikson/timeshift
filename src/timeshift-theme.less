@import url('https://fonts.googleapis.com/css2?family=Lato');

@font-name: 'Lato';

@primary-color: #1b1919;
@font-family: @font-name;
@primary-background-color: #ffffff;

// Typography
@text-color: #001122;

// Locations list
@location-item-bg: RGBA(246, 246, 246, 0.2);
@location-item-bg-hover: RGBA(246, 246, 246);
@location-item-border: RGBA(0, 17, 34, 0.1);
@location-item-name: #001122;
@location-edit-icon: #FFFFFF;

@modal-ok-button-bg: #37B24D;
@modal-ok-button-color: #ffffff;
