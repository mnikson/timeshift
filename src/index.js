/* eslint-disable react/jsx-filename-extension */
// Libs
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import dotenv from 'dotenv';

import config from './config';

// Components
import App from './components/App';

// Globals
import createStore from './globals/store';

import reportWebVitals from './reportWebVitals';

dotenv.config();

const { store } = createStore();

ReactDOM.render(
  <Provider store={ store }>
    <App />
  </Provider>,
  document.getElementById(config.rootElement)
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
