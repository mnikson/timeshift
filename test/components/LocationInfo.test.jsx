/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';

// Components
import { LocationInfo } from '../../src/components';

// Mock
import { data } from '../__mock__/locationsData';

describe('LocationInfo component tests', () => {
  it('should render component', () => {
    const location = data[0];
    const viewsCount = 20;

    render(
      <LocationInfo
        date={ location.createdAt }
        name={ location.name }
        views={ viewsCount }
        description={ location.description }
        userCount={ location.userCount }
      />
    );
    const name = screen.getByText(location.name);
    expect(name).toBeInTheDocument();
    const description = screen.getByText(location.description);
    expect(description).toBeInTheDocument();
    const userCount = screen.getByText(`${location.userCount} Users`);
    expect(userCount).toBeInTheDocument();
    const views = screen.getByText(`${viewsCount} Views`);
    expect(views).toBeInTheDocument();
  });

  it('should render component without name', () => {
    const location = data[0];
    const viewsCount = 20;

    render(
      <LocationInfo
        date={ location.createdAt }
        views={ viewsCount }
        description={ location.description }
        userCount={ location.userCount }
      />
    );
    const name = screen.queryByText(location.name);
    expect(name).toBeNull();
  });

  it('should render component without description', () => {
    const location = data[0];
    const viewsCount = 20;

    const { container } = render(
      <LocationInfo
        name={ location.name }
        description={ location.description }
        date={ location.createdAt }
        views={ viewsCount }
        userCount={ location.userCount }
      />
    );

    expect(container).toMatchSnapshot();
  });
});
