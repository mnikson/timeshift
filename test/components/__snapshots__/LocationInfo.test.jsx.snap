// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationInfo component tests should render component without description 1`] = `
<div>
  <div>
    <h2
      data-testid="location-name-undefined"
    >
      McGlynnview
    </h2>
    <div>
      6221
       Users
    </div>
    <div>
      10:10 am (+01:00)
    </div>
    <div
      data-testid="location-views-undefined"
    >
      20 Views
    </div>
    <div>
      <div>
        Description
      </div>
      <p>
        Investment Account
      </p>
    </div>
  </div>
</div>
`;
